examples/src/main/java/com/mapbox/navigation/examples/PermissionsHelper.kt

```diff
@@ -0,0 +1,118 @@
+            permissions.add(BACKGROUND_LOCATION_PERMISSION)
+        }
+        requestPermissions(activity, permissions.toTypedArray())
+    }
+
+    private fun requestPermissions(activity: Activity, permissions: Array<String>) {
+        val permissionsToExplain = ArrayList<String>()
+        for (permission in permissions) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(activity, permission)) {
+                permissionsToExplain.add(permission)
+            }
+        }
+        if (permissionsToExplain.isNotEmpty()) {
+            listener?.onExplanationNeeded(permissionsToExplain)
+        }
+        ActivityCompat.requestPermissions(
+            activity, permissions,
+            LOCATION_PERMISSIONS_REQUEST_CODE
+        )
+    }
+
+    /**
+     * You should call this method from your activity onRequestPermissionsResult.
+     *
+     * @param requestCode The request code passed in requestPermissions(android.app.Activity, String[], int)
+     * @param permissions The requested permissions. Never null.
+     * @param grantResults The grant results for the corresponding permissions which is either
+     * PERMISSION_GRANTED or PERMISSION_DENIED. Never null.
+     */
+    fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        when (requestCode) {
+            LOCATION_PERMISSIONS_REQUEST_CODE -> listener?.let {
+                val granted = grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED
+                it.onPermissionResult(granted)
+            }
+            else -> {
+            }
+        }
+    }
+
+    companion object {
+
+        private fun isPermissionGranted(context: Context, permission: String): Boolean {
+            return (ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED)
+        }
+
+        private fun isCoarseLocationPermissionGranted(context: Context): Boolean {
+            return isPermissionGranted(context, COARSE_LOCATION_PERMISSION)
+        }
+
+        private fun isFineLocationPermissionGranted(context: Context): Boolean {
+            return isPermissionGranted(context, FINE_LOCATION_PERMISSION)
+        }
+
+        private fun isBackgroundLocationPermissionGranted(context: Context): Boolean {
+            return isPermissionGranted(context, BACKGROUND_LOCATION_PERMISSION)
+        }
+
+        fun areLocationPermissionsGranted(context: Context): Boolean {
+            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                isBackgroundLocationPermissionGranted(context) ||
+                        isCoarseLocationPermissionGranted(context) ||
+                        isFineLocationPermissionGranted(context)
+            } else {
+                isCoarseLocationPermissionGranted(context) || isFineLocationPermissionGranted(context)
+            }
```

Comment:
return 
isCoarseLocationPermissionGranted(context) || 
           isFineLocationPermissionGranted(context) ||
           (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q && isBackgroundLocationPermissionGranted(context))

Reply:
btw, we don't even need this `...VERSION_CODES.Q` check because it's not possible on Android 10 to get only background permissions without getting `Fine/Coarse Location`